subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -0,0 +1,131 @@
+include::sample[dir="samples/java/jvm-multi-project-with-toolchains/kotlin/",files="buildSrc/src/main/kotlin/myproject.java-conventions.gradle.kts[tags=toolchain]"]
+====
+
+Executing the build (e.g. using `gradle check`) will now handle several things for you and others running your build
+
+1. Setup all compile and test tasks to use the defined toolchain and not the JVM that Gradle is running with
+2. Gradle detects <<sec:auto_detection,locally installed JVMs>>
+3. Gradle chooses a JRE/JDK matching the requirements of the build (in this case a JVM supporting Java 14)
+4. If no matching JVM is found, it will automatically download a matching JDK from https://adoptopenjdk.net/[AdoptOpenJDK]
+
+== Specify custom toolchains for individual tasks
+
+In case you want to tweak which toolchain is used for a specific task, you can specify the exact tool a task is using.
+For example, a `Test` task exposes a `JavaLauncher` property that defines which java executable to use to launch the tests.
+
+In the example below, we configure the default toolchain to use JDK8 for compiling and testing.
+Additionally, we introduce a new Test task that is going to run our unit tests but using a JDK 14. Depending on the task, a JRE might be enough while for other tasks (e.g. compilation), a JDK is required.
+By default, Gradle prefers installed JDKs over JREs.
+
+Toolchains tool providers can be obtained from the link:{javadocPath}/org/gradle/jvm/toolchain/JavaToolchainService.html[`javaToolchains`] extension.
+
+Three tools are available:
+
+* A `JavaCompiler` which is the tool used by the `JavaCompile` task
+* A `JavaLauncher` which is the tool used by the `JavaExec` or `Test` tasks
+* A `JavadocTool` which is the tool used by the `Javadoc` task
+
+====
+include::sample[dir="samples/java/jvm-multi-project-with-toolchains/groovy/",files="list/build.gradle[tags=customToolchain]"]
+include::sample[dir="samples/java/jvm-multi-project-with-toolchains/kotlin",files="list/build.gradle.kts[tags=customToolchain]"]
+====
+
+[[sec:auto_detection]]
+== Auto detection of installed toolchains
+
+By default, Gradle automatically detects local JRE/JDK installations so no further configuration is required by the user.
+The following is a list of common package managers and locations that are supported by the JVM auto detection.
+
+Operation-system specific locations:
+
+* Linux
+* MacOs
+* Windows
+
+Supported Package Managers:
+
+* https://asdf-vm.com/#/[Asdf-vm]
+* https://github.com/shyiko/jabba[Jabba]
+* https://sdkman.io/[SDKMAN!]
+
+=== How to disable auto-detection
+
+In order to disable auto-detection, you can use the `org.gradle.java.installations.auto-detect` gradle property.
+Either start gradle using `-Porg.gradle.java.installations.auto-detect=false` or put
+`org.gradle.java.installations.auto-detect=false` into your `gradle.properties` file.
+
+[[sec:provisioning]]
+== Auto Provisioning
```

Comment:
:o: Can auto provisioning be disabled? I can imagine environments where you want to fail if a compatible toolchain is not installed locally.

Reply:
Yes, added already ... though the PR still needs to be merged 😉